tests/Unleash.Tests/Communication/UnleashApiClient_Features_Backoff_Tests.cs

```diff
@@ -0,0 +1,263 @@
+        );
+
+        // Act
+        var result = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+        var result2 = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+        
+        // Assert
+        messageHandler.CallCount.Should().Be(2);
+    }
+
+    [Test]
+    public void Should_Skip_One_After_First_429()
+    {
+        // Arrange
+        var messageHandler = new CountingConfigurableHttpMessageHandler
+        (
+            new List<HttpResponseMessage>
+            {
+                TooManyRequests,
+                Ok
+            }
+        );
+        var apiClient = GetClient(messageHandler);
+
+        // Act 1
+        var result = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+
+        // Assert 1
+        messageHandler.CallCount.Should().Be(1);
+
+        // Act 2
+        var result2 = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+
+        // Assert 2
+        messageHandler.CallCount.Should().Be(1);
+
+        // Act 3
+        var result3 = Task.Run(() => apiClient.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+
+        // Assert 3
+        messageHandler.CallCount.Should().Be(2);
+    }
+
+    [Test]
+    public void Backoff_Caps_Out_At_10() {
+        // Arrange
+        var messageHandler = new CountingConfigurableHttpMessageHandler
+        (
+            new List<HttpResponseMessage>
+            {
+                TooManyRequests,
+                TooManyRequests,
+                TooManyRequests,
+                TooManyRequests,
+                TooManyRequests,
+                TooManyRequests,
+                TooManyRequests,
+                TooManyRequests,
+                TooManyRequests,
+                TooManyRequests,
+                Ok,
+                Ok
+            }
+        );
+
+        var client = GetClient(messageHandler);
+
+        // Act
+        for (var i = 0; i < 55; i++)
+        {
+            var loopResult = Task.Run(() => client.FetchToggles("etag", CancellationToken.None))
+                .GetAwaiter()
+                .GetResult();
+        }
+
+        for (var i = 0; i < 10; i++)
+        {
+            var loopResult = Task.Run(() => client.FetchToggles("etag", CancellationToken.None))
+                .GetAwaiter()
+                .GetResult();
+        }
+
+        messageHandler.CallCount.Should().Be(10);
+
+        var resultAfter = Task.Run(() => client.FetchToggles("etag", CancellationToken.None))
+            .GetAwaiter()
+            .GetResult();
+
+        // Assert
+        // Calls on the 11th retry
+        messageHandler.CallCount.Should().Be(11);
+
+        // Another loop of 9
```

Comment:
Up to you but I'd yeet this comment. Hopefully anyone reading this knows how loops work

```suggestion
```